src/ecdsa.rs

```diff
@@ -0,0 +1,105 @@
+use crate::bindings::{
+    secp256k1_ec_pubkey_create, secp256k1_ecdsa_sign, secp256k1_ecdsa_signature,
+    secp256k1_ecdsa_verify, secp256k1_pubkey,
+};
+use crate::context::Context;
+use crate::scalar::Scalar;
+
+#[derive(Debug)]
+/// Errors in ECDSA signature operations
+pub enum Error {
+    /// Error occurred due to invalid secret key
+    InvalidSecretKey,
+}
+
+/**
+Signature is a wrapper around libsecp256k1's internal secp256k1_pubkey struct.
```

Comment:
s/Signature/PubKey/

Also these aren't internal structs, unlike everything else in this crate these are the actual public structs/functions